web3/_utils/datatypes.py

```diff
@@ -16,12 +24,25 @@ def verify_attr(class_name, key, namespace):
 
 
 class PropertyCheckingFactory(type):
-    def __init__(cls, name, bases, namespace, **kargs):
+    def __init__(
+        cls,
+        name: str,
+        bases: Tuple[type],
```

Comment:
This means a length-1 tuple with a single type.  I think you probably mean: `Tuple[Type[Any], ...]` which may-or-may-not be equivalent to `Tuple[type, ...]`